plugins/catalog-backend/src/ingestion/processors/GitLabDiscoveryProcessor.ts

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { LocationSpec } from '@backstage/catalog-model';
+import { Config } from '@backstage/config';
+import { ScmIntegrations } from '@backstage/integration';
+import { Logger } from 'winston';
+import * as results from './results';
+import { CatalogProcessor, CatalogProcessorEmit } from './types';
+import { GitLabClient, GitLabProject, paginated } from './gitlab';
+import {
+  CacheClient,
+  CacheManager,
+  PluginCacheManager,
+} from '@backstage/backend-common';
+
+/**
+ * Extracts repositories out of an GitLab instance.
+ */
+export class GitLabDiscoveryProcessor implements CatalogProcessor {
+  private readonly integrations: ScmIntegrations;
+  private readonly logger: Logger;
+  private readonly cache: CacheClient;
+
+  static fromConfig(config: Config, options: { logger: Logger }) {
+    const integrations = ScmIntegrations.fromConfig(config);
+    const pluginCache =
+      CacheManager.fromConfig(config).forPlugin('gitlab-discovery');
+
+    return new GitLabDiscoveryProcessor({
+      ...options,
+      integrations,
+      pluginCache,
+    });
+  }
+
+  private constructor(options: {
+    integrations: ScmIntegrations;
+    pluginCache: PluginCacheManager;
+    logger: Logger;
+  }) {
+    this.integrations = options.integrations;
+    this.cache = options.pluginCache.getClient();
+    this.logger = options.logger;
+  }
+
+  async readLocation(
+    location: LocationSpec,
+    _optional: boolean,
+    emit: CatalogProcessorEmit,
+  ): Promise<boolean> {
+    if (location.type !== 'gitlab-discovery') {
+      return false;
+    }
+
+    const { group, host, branch, catalogPath } = parseUrl(location.target);
+
+    const integration = this.integrations.gitlab.byUrl(`https://${host}`);
+    if (!integration) {
+      throw new Error(
+        `There is no GitLab integration that matches ${host}. Please add a configuration entry for it under integrations.gitlab`,
+      );
+    }
+
+    const client = new GitLabClient({
+      config: integration.config,
+      logger: this.logger,
+    });
+    const startTimestamp = Date.now();
+    this.logger.info(`Reading GitLab projects from ${location.target}`);
```

Comment:
I know the GitHub discovery processor does logger.info the same thing but is it really valuable to log here? 
Personally I think the second logline saying duration and amount of things scanned is way more interesting but I would argue all of the "I'm doing something again!" should be debug level.

Reply:
Fair enough, I'll drop it down to debug. I added the log line mostly because it was also present in other processors, indeed.